transport_nantes/topicblog/models.py

```diff
@@ -518,3 +521,107 @@ def get_image_fields(self) -> list:
                 image_fields.append(field.name)
 
         return image_fields
+
+
+######################################################################
+# TopicBlogEmail
+
+class TopicBlogEmail(TopicBlogObjectBase):
+    """Represent an email.
+
+    This can be rendered as an email to be sent or as a web page that
+    the user clicks (or shares) with exactly the same content.
+
+    Note that many fields are nullable in order to permit saving
+    drafts during composition.  Publication and sending, however, must
+    validate that all necessary fields are provided.
+
+    This model must be marked immutable.  That's hard to enforce, but
+    changes must only be spelling and typographical fixes.  If we send
+    a mail, we can't have the web version change in the mean time.
+
+    The header_image only displays on the website, not in emails.  A
+    good way to make an email stay unread is to begin with an image so
+    that the user doesn't see what's coming.
+
+    """
+    class Meta:
+        permissions = (
+            # The simpleset permission allows a user to view TBEmails
+            # that are draft or retired.
+            ("tbe.may_view", "May view unpublished TopicBlogEmails"),
+
+            # Granting edit permission to users does not in itself
+            # permit them to publish or retire, so it is reasonably
+            # safe.
+            ("tbe.may_edit", "May create and modify TopicBlogEmails"),
+
+            # Finally, we can grant users permission to publish, to
+            # self-publish (implies tbe_may_publish), to self-retire,
+            # and to retire (implies self-retire).  Permission to
+            # retire implies permission to re-publish.
+            ("tbe.may_publish", "May publish TopicBlogEmails"),
+            ("tbe.may_publish_self", "May publish own TopicBlogEmails"),
+            ("tbe.may_send", "May send TopicBlogEmails"),
+            ("tbe.may_send_self", "May send own TopicBlogEmails"),
+        )
+
+    subject = models.CharField(max_length=80, blank=True)
+    header_image = models.ImageField(
+        upload_to='header/', blank=True,
+        help_text='résolution recommandée : 1600x500')
+
+    # Content #######################################################
+    body_text_1_md = models.TextField(blank=True)
```

Comment:
Agreed, but TBLauncher would have rather different content.

It occurred to me we might want an intermediate abstract base class for some of the common cases, but it seemed better to leave that to a future refactoring rather than try to predict everything all at once now.

Reply:
Yes the intermediate abstract class sounds fine to me. As you said maybe this refactoring shouldn't happen now, let's wait for the Launchers.